### ISSUE TITLE
Self-hosted runner throws Error Message: Cannot open page 130455. 

### ISSUE THREAD

**Issue by spesotskiy** [microsoft/navcontainerhelper#3215]
We switched to a self-hosted runner (Azure VM) and start getting this error frequently:
`Error Message: Cannot open page 130455. You might need to import the test toolkit to the container and/or remove the folder C:\ProgramData\BcContainerHelper\Extensions\bc6586510545\PsTestTool and retry.`
```
PowerShell Call Stack:
  at Invoke-ScriptInBcContainer, C:\ProgramData\BcContainerHelper\6.0.0\BcContainerHelper\ContainerHandling\Invoke-ScriptInNavContainer.ps1: line 71
  at Run-TestsInBcContainer, C:\ProgramData\BcContainerHelper\6.0.0\BcContainerHelper\AppHandling\Run-TestsInNavContainer.ps1: line 377
  at <ScriptBlock>, C:\ProgramData\BcContainerHelper\6.0.0\BcContainerHelper\AppHandling\Run-AlPipeline.ps1: line 827
```
On a second or third rerun it happens to disappear, but still very annoying.

I have found there was #700 and you fixed it by #708 

Now it fails in line https://github.com/microsoft/navcontainerhelper/blob/92311ffa104110645cb8d22ed531cfec29dfbea4/AppHandling/Run-TestsInNavContainer.ps1#L377

I am wondering, would it be right to always remove this folder in line 376, just before the Invoke-ScriptInBcContainer call?

If my suggestion does not make much sense, please suggest a better place, maybe in one of overridden scripts, where the removal can be done by me.

**Comment by freddydk** [comment]
Before I can answer anything, I will need to full log showing container creation and test run.

**Comment by spesotskiy** [comment]
Here are two log samples:
[logs_152.zip](https://github.com/microsoft/navcontainerhelper/files/13057406/logs_152.zip)
[logs_155.zip](https://github.com/microsoft/navcontainerhelper/files/13057407/logs_155.zip)

**Comment by freddydk** [comment]
And, could you post the logs from one of the runs which then succeeds?

**Comment by spesotskiy** [comment]
[first failed run](https://github.com/microsoft/navcontainerhelper/files/13061794/logs_140.zip)
[second failed run](https://github.com/microsoft/navcontainerhelper/files/13061796/logs_141.zip)
[successful run after 2 attempts on the same PR](https://github.com/microsoft/navcontainerhelper/files/13061793/logs_142.zip)

**Comment by freddydk** [comment]
Could you try to remove the traefik container on the self-hosted runners to see whether this is causing this instability?

**Comment by spesotskiy** [comment]
I have got another VM in hope it will help me escape this issue, but it still here. Now I own this VM so can do whatever I want.
On a new VM I ran PR build workflow and then CI\CD workflow. 
The PR Build has passed
[logs_169.zip](https://github.com/microsoft/navcontainerhelper/files/13078983/logs_169.zip)
 , but CI\CD has failed as always on starting running tests.
[logs_170.zip](https://github.com/microsoft/navcontainerhelper/files/13078984/logs_170.zip)

What is the traefik container and how to remove it? I would do it immediately.

**Comment by freddydk** [comment]
In the two above dumps, traefik is not enabled, so my theory of blaming traefik isn't right.
I don't know what is wrong with your runners.
I always use https://aka.ms/getbuildagent to create Azure VMs with runners and have never seen this problem.

**Comment by freddydk** [comment]
How frequent is the problem?
1 out of 10 runs - or 1 out of 2?
How many VMs do you have and how many runners exists on each VM?

**Comment by spesotskiy** [comment]
Now I have one VM and one runner on it. It was not me who created this runner. Though the same issue I experienced first on another VM. Will verify if they were created weirdly.
Today I tried several times and always was always stopped by this error. So, it is 100% today.

By the way, I have tested the fix (remove the folder before running tests), I had suggested. It does not work, because the function Run-Tests is declared by a script in that removed folder.
So, the message talking about either the missing page 130455 or not removed PsTestTool folder most likely caused by the missing page. 
It fails here:
https://github.com/microsoft/navcontainerhelper/blob/c4ca9476e05e6c6482eeb5f02bc8f6917e7eb08c/AppHandling/PsTestFunctions.ps1#L535
Though I see that the test toolkit seems to be successfully published all the time. 
How could the page disappear?
Maybe something wrong with $clientContext?

Will continue experimenting.

**Comment by freddydk** [comment]
I don't think the page is missing - maybe it is a timing issue.
It could be a timing issue.
If you place a script called RunTestsInBcContainer.ps1 in the .AL-Go folder with the following content:

```powershell
Param([Hashtable]$parameters)
Start-Sleep -seconds 5
Run-TestsInBcContainer @parameters
```

**Comment by spesotskiy** [comment]
After 5 seconds delay (as you suggested above) before Run-TestsInBcContainer it has failed again with the same error. :(
Interesting that I got one PR in without that error today. Will compare logs to see a difference between these two runs.

**Comment by spesotskiy** [comment]
I don't know how I overlooked this discussion https://github.com/microsoft/navcontainerhelper/issues/2902

According to it I simply modified .github/AL-Go-Settings.json by adding a property:
 `"DefaultNewContainerParameters": {"isolation": "hyperv"}`
Now I enjoy stable PR workflow runs. 

Though I am not fully happy with the time spent on them. I have an assumption that when isolation was different it took by 10-15 minutes less. 
Will investigate further. 

As discussed in https://github.com/microsoft/navcontainerhelper/issues/2902 that is a possibility to resolve the problem by installing dotNet6 on the host. Will try it.
